plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaFilterManager.java

```diff
@@ -140,10 +142,17 @@ public KafkaFilteringResult getKafkaFilterResult(
 
             // push down partitions
             final Set<Long> finalPartitionIdsFiltered = partitionIdsFiltered;
-            List<PartitionInfo> partitionFilteredInfos = partitionInfos.stream()
+            partitionInfos = partitionInfos.stream()
                     .filter(partitionInfo -> finalPartitionIdsFiltered.contains((long) partitionInfo.partition()))
                     .collect(toImmutableList());
-            return new KafkaFilteringResult(partitionFilteredInfos, partitionBeginOffsets, partitionEndOffsets);
+        }
+
+        // push down limit
+        if (limit.isPresent()) {
+            Map<TopicPartition, Long> finalPartitionBeginOffsets = partitionBeginOffsets;
+            partitionEndOffsets = overridePartitionEndOffsets(partitionEndOffsets,
+                    partition -> (finalPartitionBeginOffsets.get(partition) != INVALID_KAFKA_RANGE_INDEX)
+                            ? Optional.of(finalPartitionBeginOffsets.get(partition) + limit.getAsLong()) : Optional.empty());
```

Comment:
Please correct me if I am wrong.. this would work only if the each message is on size 1 byte then we might assume that if we add `limit` to it... If each message is of different sizes then the offset might differ

Reply:
Kafka offset is the metadata of Kafka and the inherent information of each message. 
According to my understanding, it should have nothing to do with the message size ? I don't understand the limitations of byte ?